bindings/pyroot/src/Converters.cxx

```diff
@@ -1013,24 +1013,54 @@ Bool_t PyROOT::TValueCppObjectConverter::SetArg(
 Bool_t PyROOT::TRefCppObjectConverter::SetArg(
       PyObject* pyobject, TParameter& para, TCallContext* /* ctxt */ )
 {
-   if ( ! ObjectProxy_Check( pyobject ) )
-      return kFALSE;
+   if ( ObjectProxy_Check( pyobject ) ) {   // It is PyROOT object
+      ObjectProxy* pyobj = (ObjectProxy*)pyobject;
+      if ( pyobj->ObjectIsA() && Cppyy::IsSubtype( pyobj->ObjectIsA(), fClass ) ) {
+      // calculate offset between formal and actual arguments
+            para.fValue.fVoidp = pyobj->GetObject();
+            if ( pyobj->ObjectIsA() != fClass ) {
+            para.fValue.fLong += Cppyy::GetBaseOffset(
+                  pyobj->ObjectIsA(), fClass, para.fValue.fVoidp, 1 /* up-cast */ );
+            }
 
-   ObjectProxy* pyobj = (ObjectProxy*)pyobject;
-   if ( pyobj->ObjectIsA() && Cppyy::IsSubtype( pyobj->ObjectIsA(), fClass ) ) {
-   // calculate offset between formal and actual arguments
-      para.fValue.fVoidp = pyobj->GetObject();
-      if ( pyobj->ObjectIsA() != fClass ) {
-         para.fValue.fLong += Cppyy::GetBaseOffset(
-            pyobj->ObjectIsA(), fClass, para.fValue.fVoidp, 1 /* up-cast */ );
+            para.fTypeCode = 'V';
+            return kTRUE;
+      } else if ( ! TClass::GetClass( Cppyy::GetFinalName( fClass ).c_str() )->GetClassInfo() ) {
+      // assume "user knows best" to allow anonymous reference passing
+            para.fValue.fVoidp = pyobj->GetObject();
+            para.fTypeCode = 'V';
+            return kTRUE;
+      }
+   }
+   else if ( PyTuple_Check( pyobject ) ){  // It is a Python tuple (equivalent to a C++ initializer list)
+
+      // instantiate an object proxy of this class
+      if( ! fObjProxy ) {
+      // retrieve python class
+         PyObject* pyclass = CreateScopeProxy( fClass );
+         if ( ! pyclass ) return kFALSE;                  // error has been set in CreateScopeProxy
+         fObjProxy = (ObjectProxy*)((PyTypeObject*)pyclass)->tp_new( (PyTypeObject*)pyclass, pyobject, NULL );
```

Comment:
Do we need to pass the tuple (pyobject) when we construct the proxy here?

Reply:
No, it not needed. The last two arguments are ignored. We can call it as `tp_new( (PyTypeObject*)pyclass, NULL, NULL )`